### PR TITLE
External probes now return multiple items to collect

### PR DESCRIPTION
**Pull request by cpop02**
External probes can not return multiple items to collect.

### CODE REVIEW

File: src/OVAL/probes/independent/environmentvariable_probe.c [hunk 1]
```diff
@@ -84,88 +84,77 @@ static int read_environment(SEXP_t *un_ent, probe_ctx *ctx)
...
+    if(eval == NULL || eval->environment_variable_probe == NULL) {
+        ret = PROBE_EOPNOTSUPP;
+        goto fail;
+    }
+    in = probe_ctx_getobject(ctx);
+    id = probe_obj_getattrval(in, "id");
+    if(id == NULL) {
+        ret = PROBE_ENOVAL;
+        goto fail;
+    }
```
Comment (cs-sotescu) [inline src/OVAL/probes/independent/environmentvariable_probe.c:109]: This part is missing from the new implementation. It's mandatory to collect only the env vars matching the name provided by `un_ent`.

File: src/OVAL/probes/independent/environmentvariable58_probe.c [hunk 2]
```diff
@@ -402,111 +402,99 @@ static int read_environment(SEXP_t *pid_ent, SEXP_t *name_ent, probe_ctx *ctx)
...
-        }
-
-        env_name = SEXP_string_newf("%s", name);
-        env_value = SEXP_string_newf("%s", oval_external_probe_value_get_string(val));
-        if (probe_entobj_cmp(un_ent, env_name) == OVAL_RESULT_TRUE) {
-            dI("EXTPROBE: environmentvariable: matched env_name=%s, value=%s", name, val);
-            item = probe_item_create(OVAL_INDEPENDENT_ENVIRONMENT_VARIABLE, NULL, "name", OVAL_DATATYPE_SEXP, env_name, "value", OVAL_DATATYPE_SEXP,
-                                     env_value, NULL);
-            probe_item_collect(ctx, item);
-        }
```
Comment (cs-sotescu) [inline src/OVAL/probes/independent/environmentvariable58_probe.c:427]: This part is missing from the new implementation. It's mandatory to collect only the env vars matching the name provided by un_ent.

File: src/OVAL/probes/independent/environmentvariable_probe.c (hunk 1 above)
Comment (cpop02) [reply src/OVAL/probes/independent/environmentvariable_probe.c:109]: I've left it to the external probe to interpret the query and return only the matching env variables. I'm not sure what approach is better, so, if you think it's better to implement/readd that logic here, I'll do it.

File: src/OVAL/probes/independent/environmentvariable58_probe.c (hunk 2 above)
Comment (cpop02) [reply src/OVAL/probes/independent/environmentvariable58_probe.c:427]: I've readded the filtering functionality.

File: src/OVAL/probes/independent/environmentvariable_probe.c (hunk 1 above)
Comment (cpop02) [reply src/OVAL/probes/independent/environmentvariable_probe.c:109]: I've readded the filtering functionality.

Comment (cpop02) [reply src/OVAL/probes/independent/environmentvariable_probe.c:109]: I've backed up the new commits on a new branch and reverted the changes. The specialised external probes will be removed in a future PR.

File: src/OVAL/probes/independent/environmentvariable58_probe.c (hunk 2 above)
Comment (cpop02) [reply src/OVAL/probes/independent/environmentvariable58_probe.c:427]: I've backed up the new commits on a new branch and reverted the changes. The specialised external probes will be removed in a future PR.

**Review by sorinescu — APPROVED** [review]

**Pull request merged**